O que aprendemos na aula passada

    Importar a biblioteca Pandas;
    Realizar a leitura de arquivos csv;
    Identificar um DataFrame;
    Visualizar linhas iniciais e finais de um DataFrame;
  

    Exercício: Entre no Kaggle , escolha um dataset e faça os procedimentos acima.

O que faremos agora?

  -  Verificar a quantidade de linhas e colunas de um DataFrame;
  -  Conferir os tipos de dados de cada coluna de um DataFrame;
  -  Selecionar uma ou múltiplas colunas.


In [6]:
import pandas as pd





In [5]:
url = "dataset/aluguel.csv"
pd.read_csv(url)

Tipo;Bairro;Quartos;Vagas;Suites;Area;Valor;Condominio;IPTU
0              Quitinete;Copacabana;1;0;0;40;1700;500;60         
1                  Casa;Jardim Botânico;2;0;1;100;7000;;         
2      Conjunto Comercial/Sala;Barra da Tijuca;0;4;0;...         
3                 Apartamento;Centro;1;0;0;15;800;390;20         
4             Apartamento;Higienópolis;1;0;0;48;800;230;         
...                                                  ...         
32955               Quitinete;Centro;0;0;0;27;800;350;25         
32956       Apartamento;Jacarepaguá;3;1;2;78;1800;800;40         
32957  Apartamento;São Francisco Xavier;2;1;0;48;1400...         
32958              Apartamento;Leblon;2;0;0;70;3000;760;         
32959  Conjunto Comercial/Sala;Centro;0;0;0;250;6500;...         

[32960 rows x 1 columns]

In [8]:
dados = pd.read_csv(url,sep=";")
dados

Tipo                Bairro  Quartos  Vagas  Suites  \
0                    Quitinete            Copacabana        1      0       0   
1                         Casa       Jardim Botânico        2      0       1   
2      Conjunto Comercial/Sala       Barra da Tijuca        0      4       0   
3                  Apartamento                Centro        1      0       0   
4                  Apartamento          Higienópolis        1      0       0   
...                        ...                   ...      ...    ...     ...   
32955                Quitinete                Centro        0      0       0   
32956              Apartamento           Jacarepaguá        3      1       2   
32957              Apartamento  São Francisco Xavier        2      1       0   
32958              Apartamento                Leblon        2      0       0   
32959  Conjunto Comercial/Sala                Centro        0      0       0   

       Area   Valor  Condominio    IPTU  
0        40  1700.0       500.0    60.0  
1       100  7000.0         NaN     NaN  
2       150  5200.0      4020.0  1111.0  
3        15   800.0       390.0    20.0  
4        48   800.0       230.0     NaN  
...     ...     ...         ...     ...  
32955    27   800.0       350.0    25.0  
32956    78  1800.0       800.0    40.0  
32957    48  1400.0       509.0    37.0  
32958    70  3000.0       760.0     NaN  
32959   250  6500.0      4206.0  1109.0  

[32960 rows x 9 columns]

Como retorno, obtemos a tupla (32960, 9), com dois elementos. O primeiro corresponde à quantidade de linhas; o segundo, à de colunas. Sendo assim, temos 32960 linhas e 9 colunas

In [9]:
dados.shape

(32960, 9)

Veremos, agora, quais são as colunas:

In [10]:
dados.columns

Index(['Tipo', 'Bairro', 'Quartos', 'Vagas', 'Suites', 'Area', 'Valor',
       'Condominio', 'IPTU'],
      dtype='object')

    Tipo: tipo de imóvel;
    Bairro: bairro em que está localizado;
    Quartos: número de quartos;
    Vagas: número de vagas de garagem;
    Suites: número de suítes;
    Area: área do imóvel em metros quadrados;
    Valor: valor do imóvel;
    Condominio: valor mensal do condomínio;
    IPTU: valor do IPTU.

Com isso, podemos inferir o tipo de dado de cada coluna, mas é importante nos certificar para garantir que estejam coerentes. Há várias formas de acessar esta tipagem, mas, aqui, usaremos o método info():

In [11]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32960 entries, 0 to 32959
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Tipo        32960 non-null  object 
 1   Bairro      32960 non-null  object 
 2   Quartos     32960 non-null  int64  
 3   Vagas       32960 non-null  int64  
 4   Suites      32960 non-null  int64  
 5   Area        32960 non-null  int64  
 6   Valor       32943 non-null  float64
 7   Condominio  28867 non-null  float64
 8   IPTU        22723 non-null  float64
dtypes: float64(3), int64(4), object(2)
memory usage: 2.3+ MB


Comecemos pela coluna "Tipo", que possui dados do tipo object. Para selecionar uma única coluna, basta passar o nome do dataframe e o nome da coluna entre colchetes:

In [12]:
dados['Tipo']

0                      Quitinete
1                           Casa
2        Conjunto Comercial/Sala
3                    Apartamento
4                    Apartamento
                  ...           
32955                  Quitinete
32956                Apartamento
32957                Apartamento
32958                Apartamento
32959    Conjunto Comercial/Sala
Name: Tipo, Length: 32960, dtype: object

Vejamos uma coluna dos tipos int e float. Faremos a consulta das duas juntas, da seguinte forma:

In [13]:
dados[['Quartos', 'Valor']]

Quartos   Valor
0            1  1700.0
1            2  7000.0
2            0  5200.0
3            1   800.0
4            1   800.0
...        ...     ...
32955        0   800.0
32956        3  1800.0
32957        2  1400.0
32958        2  3000.0
32959        0  6500.0

[32960 rows x 2 columns]

: 

Faça agora o mesmo com o dataset que você baixou do Kaggle.

Revisão Git e GitHub
Pesquise a diferença entre tupla, lista e dicionários
Pesquise no notebook do Alex.